record_platform_interface/lib/src/types/record_config.dart

```diff
@@ -51,6 +51,9 @@ class RecordConfig {
   /// Android specific configuration.
   final AndroidRecordConfig androidConfig;
 
+  /// iOS specific audioCategories
+  final List<IosAudioCategories> iosAudioCategories;
```

Comment:
We should wrap those iOS specific categories in `iosConfig` of type `IosRecordConfig`.